src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -283,6 +294,62 @@ public void InitializeService(ServiceHost serviceHost, SqlToolsContext context)
 
         #region Request Handlers
 
+        /// <summary>
+        /// Completion extension load request callback
+        /// </summary>
+        /// <param name="param"></param>
+        /// <param name="requestContext"></param>
+        /// <returns></returns>
+        internal async Task HandleCompletionExtLoadRequest(CompletionExtensionParams param, RequestContext<bool> requestContext)
+        {
+            int extCount = 0;
+            try
+            {
+                //register the new assembly
+                var serviceProvider = (ExtensionServiceProvider)ServiceHostInstance.ServiceProvider;
+                var assemblies = new Assembly[] { AssemblyLoadContext.Default.LoadFromAssemblyPath(param.Assembly) };
+                serviceProvider.AddAssembliesToConfiguration(assemblies);
+                foreach (var provider in serviceProvider.GetServices<ICompletionExtensionProvider>())
+                {
+                    if (completionExtensionProviders.Contains(provider))
+                    {
+                        //skipping completion providers that already loaded
+                        continue;
+                    }
+                    completionExtensionProviders.Add(provider);
+
+                    var cancellationTokenSource = new CancellationTokenSource();
+                    cancellationTokenSource.CancelAfter(ExtensionLoadingTimeout);
+                    var cancellationToken = cancellationTokenSource.Token;
+                    var ext = await provider.CreateAsync(param.Properties ?? new Dictionary<string, object>(), cancellationToken).WithTimeout(ExtensionLoadingTimeout);
+                    if (ext == null)
+                    {
+                        await requestContext.SendError("Failed to create ICompletionExtension from " + provider.GetType().FullName);
+                        return;
+                    }
+                    string extName = ext.Name;
+                    await ext.Initialize(cancellationToken).WithTimeout(ExtensionLoadingTimeout);
+                    cancellationTokenSource.Dispose();
+                    if (!string.IsNullOrEmpty(extName))
+                    {
+                        completionExtensions.AddOrUpdate(extName, ext, (_, previous) =>
+                        {
+                            previous?.Dispose();
+                            return ext;
+                        });
```

Comment:
I don't think we should be doing this like this - the dispose should be unnecessary and could possible cause issues with stuff being disposed in the middle of execution if we somehow did get to a point where this was called for already loaded extensions. The way I'd do it is check if the extension in the params is already loaded (in the map) and if it is send an error since that's unexpected behavior. If it isn't then go ahead and initialize it and add it to the map at that point.

Reply:
There is real user scenario here, in vscode if the user upgrades intellicode extension with the updated bits for the sql completion extension, it will ask mssql extension to reload the completion extension, in this case, we should reload and update the current completion extension instance.